source/distributed/tria_base.cc

```diff
@@ -163,6 +166,32 @@ namespace parallel
         return;
       }
 
+
+    {
+      // find ghost owners
+      for (typename Triangulation<dim,spacedim>::active_cell_iterator
+           cell = this->begin_active();
+           cell != this->end();
+           ++cell)
+        if (cell->is_ghost())
+          number_cache.ghost_owners.insert(cell->subdomain_id());
+
+      Assert(number_cache.ghost_owners.size() < Utilities::MPI::n_mpi_processes(mpi_communicator), ExcInternalError());
+    }
+
+    {
+      // find level ghost owners
+      for (typename Triangulation<dim,spacedim>::cell_iterator
+           cell = this->begin();
+           cell != this->end();
+           ++cell)
+        if (cell->level_subdomain_id() != numbers::artificial_subdomain_id
+            && cell->level_subdomain_id() != locally_owned_subdomain())
```

Comment:
Aren't these two conditions equivalent to checking whether the cell is ghost? Or can you only call is_ghost on active cells?